pkg/diff/run.go

```diff
@@ -57,6 +58,11 @@ func Run(opts Options) (*Diff, error) {
 		return nil, fmt.Errorf("failed to set worktree filesystem interface: %v", err)
 	}
 
+	wt.Excludes, err = gitignore.LoadGlobalPatterns(wt.Filesystem)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load global gitignore: %v", err)
+	}
+
```

Comment:
A few things:
1. We should append to `wt.Excludes`, not override it.
2. The filesystem passed in should be the OS root (not the repo path)

See the following for an example: https://github.com/apache/skywalking-eyes/blob/5dfa68f93380a5e57259faaf95088b7f133b5778/pkg/header/check.go#L84-L94